grid2op/tests/test_Opponent.py

```diff
@@ -126,5 +126,151 @@ def test_env_opp_attack_budget_ts(self):
                 assert env.opponent_budget_per_ts == 0.5
                 assert env.oppSpace.budget == 0.
 
+    def test_RandomLineOpponent_not_enough_budget(self):
+        """Tests that the attack is ignored when the budget is too low"""
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore")
+            init_budget = 3
+            with make("rte_case14_realistic",
+                      test=True,
+                      opponent_init_budget=init_budget,
+                      opponent_budget_per_ts=0.,
+                      opponent_action_class=TopologyAction,
+                      opponent_budget_class=BaseActionBudget,
+                      opponent_class=RandomLineOpponent) as env:
+                env.seed(0)
+                obs = env.reset()
+                assert env.oppSpace.budget == init_budget
+                # The opponent can attack
+                for i in range(init_budget):
+                    obs, reward, done, info = env.step(env.action_space())
```

Comment:
maybe update the "info" dict with the opponent attack (in BaseEnv) [that would be easier to test it, and could also help the participants :-) ]